devito/ir/iet/visitors.py

```diff
@@ -1112,6 +1113,46 @@ def visit_Node(self, o, ret=None):
         return ret
 
 
+class FindWithin(FindNodes):
+
+    @classmethod
+    def default_retval(cls):
+        return [], False
+
+    """
+    Like FindNodes, but given an additional parameter `within=(start, stop)`,
+    it starts collecting matching nodes only after `start` is found, and stops
+    collecting matching nodes after `stop` is found.
+    """
+
+    def __init__(self, match, start, stop=None):
+        super().__init__(match)
+        self.start = start
+        self.stop = stop
+
+    def visit(self, o, ret=None):
+        found, _ = self._visit(o, ret=ret)
+        return found
+
+    def visit_Node(self, o, ret=None):
+        if ret is None:
+            ret = self.default_retval()
+        found, flag = ret
+
+        if o is self.start:
+            flag = True
+
+        if flag and self.rule(self.match, o):
+            found.append(o)
+        for i in o.children:
+            found, flag = self._visit(i, ret=(found, flag))
```

Comment:
Should probably exit if flag switch from true to false. Or it's gonnavisit everything afterward for nothing.

Reply:
but then I'd have to add logic to keep track that a switch from False to True had already happened (that is, that a start-stop has been encountered as opposed to not yet), which I prefer to avoid (less complexity)

Reply:
Somethign like

```
for i in o.children:
            found, newflag = self._visit(i, ret=(found, flag))
             if flag and not newflag:
                 return found, newflag
             else:
                 flag = newflag
```

Should do it and avoid quite a lot of visit